eth/client.go

```diff
@@ -792,7 +794,7 @@ func (c *client) Reward() (*types.Transaction, error) {
 		return nil, errors.Wrapf(err, "unable to get transcoder pool max size")
 	}
 
-	hints := simulateTranscoderPoolUpdate(addr, reward.Add(reward, ep.TotalStake), transcoders, len(transcoders) == int(maxSize.Int64()))
+	hints := simulateTranscoderPoolUpdate(addr, reward.Add(reward, tr.DelegatedStake), transcoders, len(transcoders) == int(maxSize.Int64()))
```

Comment:
Previously, we used the `TotalStake` field on the earnings pool here. There were two problems with this:

1. If the orchestratrator's `LastActiveStakeUpdateRound` is before the current round, then the `TotalStake` field would be 0 and empty hints would be returned which results in the increased gas usage described in #2029 
2. Even if the correct earnings pool was fetched, the `TotalStake` field on the earnings pool would reflect the orchestrator's active total stake which remains fixed throughout the current round. If the orchestrator receives delegation during the round, its active total stake would not change, but its total stake *would* change and its total stake is the value that should be used for `simulateTranscoderPoolUpdate`

Both 1 and 2 are resolved by using the `DelegatedStake` field on the `Transcoder` struct which accurately reflects the orchestrator's total stake at all points in the round.